flake.nix

```diff
@@ -0,0 +1,53 @@
+{
+  description = "Flake for Plutus";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=7d71001b796340b219d1bfa8552c81995017544a";
+    haskell-nix.url = "github:input-output-hk/haskell.nix?rev=87084d65a476cc826a0e8c5d281d494254f5bc7a";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, flake-utils, ... }:
+    (flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+      let
+        #
+        # Obtain all niv sources extending it with the 'nixpkgs' flake input
+        #
+        sources =
+          let
+            sourcesInfo = builtins.fromJSON (builtins.readFile ./nix/sources.json);
+            fetch = sourceInfo: builtins.fetchTarball { inherit (sourceInfo) url sha256; };
+          in
+          builtins.mapAttrs (name: info: fetch info) sourcesInfo // { inherit nixpkgs; };
+
+        #
+        # all packages from nix/default.nix
+        #
+        plutusPackages = import ./nix {
+          inherit system sources;
+          haskellNixOverlays = [ haskell-nix.overlay ];
+        };
+
+        #
+        # all packages from ./default.nix
+        #
+        topLevel = import ./. {
+          inherit system;
+          packages = plutusPackages;
```

Comment:
Oh I see, you're effectively doing this to override the sources? Can't we just plumb that up to the top-level like with everything else?

Reply:
We want `default.nix` which receives `nix/default.nix` which needs `sources` which we have to construct manually. Hence `sources` -> `plutusPackages` -> `topLevel`.

I really, really don't like this and `nix/default.nix` looks pretty confusing to me now. If possible i'd like to keep the ugly in the flake.nix and not so much elsewhere. Very open to all suggestions.

Reply:
Why not let `default.nix` take a `sources` argument and pass it in to `nix/default.nix`?

Reply:
I also just tried adding `sources` as argument to `default.nix` which I think is what you meant when you aid "plumb that up to the top-level" but that somehow just leads to more errors with something somewhere trying to use `builtins.currentSystem` ;/